File: packages/java/src/test/java/ai/alumnium/system/SearchTest.java
package ai.alumnium.system;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import ai.alumnium.Provider;
import org.junit.jupiter.api.Assumptions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.DisabledIf;

@DisabledIf("isHeadlessPlaywright")
class SearchTest extends BaseTest {

  @BeforeAll
  static void configure() {
    Assumptions.assumeFalse(al.model().provider() == Provider.OLLAMA, "Poor instruction following");
  }

  static boolean isHeadlessPlaywright() {
    return "playwright".equals(System.getenv("ALUMNIUM_DRIVER"))
        && !"false".equalsIgnoreCase(System.getenv("ALUMNIUM_PLAYWRIGHT_HEADLESS"));
  }

  @Test
  void searchTest() {
    navigate("https://search.brave.com");
    al.act("type 'selenium' into the search field, then press 'Enter'");
    al.check("page title contains selenium");
    assertEquals(al.get("atomic number"), 34L);
    al.check("search results contain selenium.dev");
    assertThrows(
        AssertionError.class, () -> al.check("search results do not contain selenium.dev"));
  }
}

File: packages/python/examples/pytest/search_test.py
from os import getenv

import pytest
from pytest import mark, raises

from alumnium import Provider

alumnium_driver = getenv("ALUMNIUM_DRIVER", "selenium")
playwright_headless = getenv("ALUMNIUM_PLAYWRIGHT_HEADLESS", "true")


@mark.skipif(
    alumnium_driver == "playwright" and playwright_headless == "true",
    reason="Brave Search blocks headless browsers",
)
def test_search(al, navigate):
    if al.model.provider == Provider.OLLAMA:
        pytest.xfail("Poor instruction following")

    navigate("https://search.brave.com")

    al.do("type 'selenium' into the search field, then press 'Enter'")
    al.check("page title contains selenium")

    assert al.get("atomic number") == 34

    al.check("search results contain selenium.dev")
    with raises(AssertionError):
        al.check("search results do not contain selenium.dev")
